### PR TITLE
replace(rip -> rip2): replace with maintained fork

### PR DESCRIPTION
**Pull request by TomJo2000**
Replaces the `rip` package with the maintained `rip2` fork.
I have kept the package name for continuity.
This does necessitate an epoch bump since `rip2` restarted the version numbering.

If this is not desired a new `rip2` package can be made to replace `rip` instead.

### CODE REVIEW

**Comment by Biswa96** [comment]
> If this is not desired a new `rip2` package can be made to replace `rip` instead.

I prefer the rip2 package name. For example, the exa and eza change.

According to the rip2 README, the deleted files are sent to `/tmp/graveyard-$USER` directory. In my setup, rip2 probably can not read the user name and it created graveyard-unknown directory. This does not affect the general use case, just a nitpick. Please try to reproduce it in your setup first.

Also, I wonder why the `/tmp` patch is not required now :thinking:

**Comment by TomJo2000** [comment]
> I prefer the rip2 package name. For example, the exa and eza change.

- The difference here is that the name of the binary hasn't changed,
but updating the package name is probably the better choice.
I'd like to get upstream's input on this if he's not too busy.
(CC: @MilesCranmer)

> According to the rip2 README, the deleted files are sent to `/tmp/graveyard-$USER` directory. In my setup, rip2 probably can not read the user name and it created graveyard-unknown directory. This does not affect the general use case, just a nitpick. Please try to reproduce it in your setup first.

- We could change it to `graveyard-termux` but it doesn't affect usability.
Whether it is `graveyard-u0_a123` or `graveyard-termux`

> Also, I wonder why the `/tmp` patch is not required now 🤔

- As far as I can tell `rip2` never directly references `/tmp`
See: MilesCranmer/rip2#22

**Comment by MilesCranmer** [comment]
I agree that updating the name to `rip2` probably is best as it would bring the name in line with that of cargo, (which is `cargo install rip2`) :+1:

**Comment by TomJo2000** [comment]
I have updated the build script accordingly now.

**Comment by MilesCranmer** [comment]
`rip -s` only shows files deleted from the current directory, rather than globally

**Comment by TomJo2000** [comment]
> `rip -s` only shows files deleted from the current directory, rather than globally

I deleted the `.deb` from this same directory.
I'll test this on my PC because I suspect the Termux version isn't working quite right.

**Comment by twaik** [comment]
Should not we have `TERMUX_PKG_CONFLICTS` to make sure old package is going to be deleted?

**Comment by TomJo2000** [comment]
> Should not we have `TERMUX_PKG_CONFLICTS` to make sure old package is going to be deleted?

I just copied the ones from `eza`.
Not sure why there isn't a CONFLICTS in there.

**Comment by TomJo2000** [comment]
Just managed to recreate the issue on my PC.
image
I suspect it's related to removing and restoring the same file multiple times.
I'll see if I can come up with a consistent way to reproduce the issue and send a bug report upstream.

File: packages/rip/build.sh
Comment (twaik) [inline packages/rip/build.sh:1]: We should move this one to `disabled-packages`.

Comment (TomJo2000) [reply packages/rip/build.sh:1]: It's a direct replacement.
I don't think there's any need to keep it in `disabled-packages`.
We didn't do that for `exa` either.

File: packages/rip2/build.sh
Comment (twaik) [inline packages/rip2/build.sh:10]: We need to make sure the old package is going to be removed from device, TERMUX_PKG_CONFLICT should do that, I am not sure if TERMUX_PKG_BREAKS implements this.

File: packages/rip/build.sh
Comment (twaik) [reply packages/rip/build.sh:1]: I am not so sure about that. Because this package will have different name and will be treated as separate package by CI and our repos.

Comment (twaik) [reply packages/rip/build.sh:1]: You are right about direct replacement, we do not need to move it to `disabled-packages`.

Comment (TomJo2000) [reply packages/rip/build.sh:1]: Yeah, same thing as with `exa` and `eza`.
- #17828

File: packages/rip2/build.sh
Comment (TomJo2000) [reply packages/rip2/build.sh:10]: Just a `TERMUX_PKG_REPLACES` should be enough.
That's what we did with `anacron` when that got retired last week.
https://github.com/termux/termux-packages/blob/13be95a87619afd8fd399a3386a578a772dbb38f/packages/cronie/build.sh#L16
- #21805

**Review by twaik — APPROVED** [review]

**Comment by TomJo2000** [comment]
I'd like to wait for some further developments on the upstream issue before merging this.
- MilesCranmer/rip2#54

But since it's not a Termux specific issue it wouldn't strictly be a blocker.
I just think we should have a look at it before merging here since it is quite a nasty regression.

**Comment by TomJo2000** [comment]
We've got an upstream PR.
- MilesCranmer/rip2#55

**Comment by MilesCranmer** [comment]
0.8.3 will have that fix in (thanks again for the report)

**Comment by TomJo2000** [comment]
> 0.8.3 will have that fix in (thanks again for the report)

I assume there's no need for us to backport the fix as a patch in the meantime?

**Comment by MilesCranmer** [comment]
Oh what I mean is that 0.8.3 is the patched version of 0.8. It just got released.

**Review by truboxl — APPROVED** [review]

**Pull request merged**
